Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/Importers/GltfEditorImporter.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Utilities.Gltf.Schema;
+using System.IO;
+using UnityEditor;
+using UnityEditor.Experimental.AssetImporters;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities.Gltf.Serialization.Editor
+{
+    public static class GltfEditorImporter
+    {
+        public static async void OnImportGltfAsset(AssetImportContext context)
+        {
+            var gltfAsset = (GltfAsset)ScriptableObject.CreateInstance(typeof(GltfAsset));
+            var importedObject = await GltfUtility.ImportGltfObjectFromPathAsync(context.assetPath);
+
+            gltfAsset.GltfObject = importedObject;
+            gltfAsset.name = $"{gltfAsset.GltfObject.Name}{Path.GetExtension(context.assetPath)}";
+            gltfAsset.Model = importedObject.GameObjectReference;
+            context.AddObjectToAsset("main", gltfAsset.Model);
+            context.SetMainObject(importedObject.GameObjectReference);
+            context.AddObjectToAsset("glTF data", gltfAsset);
+
+            bool reImport = false;
+
+            for (var i = 0; i < gltfAsset.GltfObject.textures.Length; i++)
+            {
+                GltfTexture gltfTexture = gltfAsset.GltfObject.textures[i];
+                var path = AssetDatabase.GetAssetPath(gltfTexture.Texture);
+
+                if (string.IsNullOrWhiteSpace(path))
+                {
+                    var textureName = gltfTexture.name;
+
+                    if (string.IsNullOrWhiteSpace(textureName))
+                    {
+                        textureName = $"Texture_{i}";
+                        gltfTexture.Texture.name = textureName;
+                    }
+
+                    context.AddObjectToAsset(textureName, gltfTexture.Texture);
+                }
+                else
+                {
+                    if (!gltfTexture.Texture.isReadable)
+                    {
+                        var textureImporter = AssetImporter.GetAtPath(path) as TextureImporter;
+                        textureImporter.isReadable = true;
+                        textureImporter.SetPlatformTextureSettings(new TextureImporterPlatformSettings { format = TextureImporterFormat.RGBA32 });
+                        textureImporter.SaveAndReimport();
+                        reImport = true;
+                    }
+                }
+            }
+
+            if (reImport)
+            {
+                var importer = AssetImporter.GetAtPath(context.assetPath);
+                importer.SaveAndReimport();
+                return;
+            }
+
+            for (var i = 0; i < gltfAsset.GltfObject.meshes.Length; i++)
```

Comment:
It looks like meshes aren't always explicitly created for the class. Do you know if its possible for this public field to end up null when deserializing? if so, we should probably have a null check for meshes

Reply:
picking up this fix